tests/utils/logging_tester.py

```diff
@@ -0,0 +1,79 @@
+from functools import wraps
+import os
+import json
+
+
+def _check_logs(reference_logfile, generated_logfile):
+    reference_logs = open(reference_logfile, "r").readlines()
+    generated_logs = open(generated_logfile, "r").readlines()
```

Comment:
Issue :warning: : these files are not being closed

Reply:
Oooooow yes thank you so much. I will correct this !

Reply:
I recommend using `with`